packages/components/src/internal/components/assay/actions.ts

```diff
@@ -81,20 +83,29 @@ export type GetProtocolOptions = {
 
 export function getProtocol(options: GetProtocolOptions): Promise<AssayProtocolModel> {
     const { copy = false, containerPath, protocolId, providerName } = options;
-    return new Promise((resolve, reject) => {
-        Ajax.request({
-            url: ActionURL.buildURL('assay', 'getProtocol.api', containerPath, {
-                copy,
-                // give precedence to the protocolId if both are provided
-                protocolId,
-                providerName: protocolId !== undefined ? undefined : providerName,
-            }),
-            success: Utils.getCallbackWrapper(data => {
-                resolve(AssayProtocolModel.create(data.data));
-            }),
-            failure: handleRequestFailure(reject, 'Failed to load assay protocol'),
+    const key = [copy, containerPath, protocolId, providerName].join('|').toLowerCase();
+
+    // Issue 49922: Cache assay protocol requests
+    if (!protocolCache[key]) {
```

Comment:
Interesting. I guess this will prevent making multiple calls for the same protocol, which will help with a page that is making these calls in different places independent of each other, but I would have thought we would cache the protocol itself (as well?).

Reply:
This is the same type of caching strategy we use for most things. Cache the request and the resolved result, the protocol in this case, is immediately returned for all subsequent callers.